### PR TITLE
feat(#72): GrowthTrendChart で3ヶ月の累積成長を可視化

### LINKED ISSUE #72: [Phase 3.3] 統計ダッシュボード実装 - KPI, チャート, アクティビティトレンド
## 概要
ユーザーの振り返り履歴から統計データを自動計算し、成長の可視化を提供します。

### Day 5-6: 統計ダッシュボード

- [ ] 統計データ集計ロジック (`src/services/analyticsService.ts`)
  - 総振り返し数
  - 今月の振り返し数
  - 連続日数計算
  - 平均文字数
  - フレームワーク使用分布
- [ ] 統計API エンドポイント (`src/app/api/analytics/`)
  - GET /api/analytics/summary - KPI サマリー
  - GET /api/analytics/trends - トレンドデータ
  - GET /api/analytics/distribution - 分布データ
- [ ] Recharts ライブラリ統合 (`npm install recharts`)
- [ ] KPI カード コンポーネント (`src/components/analytics/StatsOverview.tsx`)
- [ ] グラフ・チャート実装 (`src/components/analytics/`)
  - ReflectionFrequency.tsx - Line Chart（頻度）
  - FrameworkDistribution.tsx - Pie Chart（フレームワーク分布）
  - EmotionalTrend.tsx - Bar Chart（感情トレンド）
  - ActivityHeatmap.tsx - Heatmap（GitHub スタイル）
  - GrowthTrendChart.tsx - Area Chart（3ヶ月成長）
- [ ] ダッシュボードページ (`src/app/analytics/page.tsx`)
  - レイアウト構成
  - グラフ配置
  - レスポンシブ対応
- [ ] カスタムフック (`src/hooks/useAnalytics.ts`)

## 参考資料
- 設計書: PHASE3_DESIGN_DOCUMENT.md
- [Recharts Documentation](https://recharts.org/)

## チェックリスト
- [ ] 統計集計ロジック完了
- [ ] API エンドポイント実装完了
- [ ] Recharts 統合完了
- [ ] 全チャート表示完了
- [ ] テスト実装（統計ロジック、チャート表示）
- [ ] レスポンシブ対応完了
- [ ] コードレビュー完了

## 関連 Issue
#39

### PR DESCRIPTION
**Pull request by hiiragi17**
## Summary

issue #72 の未実装項目「GrowthTrendChart.tsx - Area Chart（3ヶ月成長）」を実装します。

統計ダッシュボードに、直近3ヶ月の累積振り返り数を Area Chart で可視化する `GrowthTrendChart` を追加しました。既存の `trends.monthly` データから派生計算するため、API 変更は不要です。

## Changes

- `src/components/statistics/GrowthTrendChart.tsx`
  - Recharts `AreaChart` で累積件数の推移を描画
  - `months` prop で表示期間をカスタマイズ可能（デフォルト 3）
  - ヘッダーに「直近 N ヶ月で X 件」のサマリー表示
  - グラデーション塗りつぶしと既存 Card UI に揃えたスタイル
- `src/components/statistics/GrowthTrendChart.test.tsx`
  - タイトル / サマリー / カスタム期間 / 空データのレンダリングを検証（4 ケース）
- `src/app/analytics/page.tsx`
  - StreakDisplay + ActivityHeatmap の下、FrameworkBreakdown の上に配置

## 関連 Issue 進捗

issue #72 のチェックリストのうち、本 PR で「GrowthTrendChart」の項目を完了します。残りの未実装項目（EmotionalTrend）は、現状 `mood` データが UI から入力されないため対象外と判断しました。

## Test plan

- [x] `pnpm vitest run src/components/statistics/GrowthTrendChart.test.tsx` が通る（4/4）
- [x] `pnpm vitest run src/components/statistics/ src/services/analyticsService.test.ts` が通る（45/45）
- [x] 追加ファイルに lint エラーなし
- [x] 追加ファイルに TypeScript エラーなし
- [ ] ブラウザで `/analytics` を開いて Area Chart が表示されることを確認
- [ ] レスポンシブ（モバイル幅）でレイアウト崩れがないことを確認

Closes #72 の一部

https://claude.ai/code/session_01CFXum283FzwBoQfh68PLAA

---
_Generated by [Claude Code](https://claude.ai/code/session_01CFXum283FzwBoQfh68PLAA)_

## Summary by CodeRabbit

* **New Features**
  * Introduced a Growth Trend Chart on the Analytics page showing cumulative totals and period-over-period growth over a configurable time window (default 3 months).
* **Tests**
  * Added test coverage verifying rendering, windowing behavior, totals calculation, edge-case handling for non-positive/fractional windows, and empty data display.

### CODE REVIEW

File: src/components/statistics/GrowthTrendChart.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+const DEFAULT_MONTHS = 3;
+
+export default function GrowthTrendChart({
+  trends,
+  months = DEFAULT_MONTHS,
+}: GrowthTrendChartProps) {
+  const data = useMemo(() => {
+    const recent = trends.monthly.slice(-months);
+    let cumulative = 0;
```
[the remark this answers is not in the document]
Comment (hiiragi17) [reply src/components/statistics/GrowthTrendChart.tsx:32]: ご指摘のとおりです、修正しました 🙏

`Math.floor` + 正の値チェックで正規化し、`0` / 負数 / 小数 / `NaN` などの不正値はデフォルト(3)にフォールバックするようにしました。`slice(-normalizedMonths)` とヘッダー表示の両方で使います。

- commit: 61a56d2
- 追加テスト: `months=0` / `months=-2` / `months=2.7` のケースをカバー

なお、別途レビュー本文でいただいた「数値単体のアサーションは誤検知しやすい」という Nitpick も同コミットで対応しています（`'6'` 等の単独 `getByText` を `'直近3ヶ月で 6 件'` のサマリー文全体マッチに変更）。

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/components/statistics/GrowthTrendChart.test.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    value: 256,
+  });
+  if (typeof globalThis.ResizeObserver === 'undefined') {
+    globalThis.ResizeObserver = class {
+      observe() {}
+      unobserve() {}
+      disconnect() {}
+    } as typeof ResizeObserver;
+  }
+});
```
[the remark this answers is not in the document]
Comment (hiiragi17) [reply src/components/statistics/GrowthTrendChart.test.tsx:54]: ご指摘ありがとうございます、修正しました 🙏

`beforeAll` で `HTMLElement.prototype.clientWidth/clientHeight` の元ディスクリプタと `globalThis.ResizeObserver` の元参照を保存し、`afterAll` で復元するようにしました。`ResizeObserver` は自前で stub した場合のみ戻すフラグも追加しています（既に存在する環境では上書きしないため、それを戻して別の実装を壊さないように）。

- commit: 21b00a9
- 既存テスト 19/19 通過

これで他テストへの副作用はなくなり、スイートが完全に隔離されます。

---
_Generated by [Claude Code](https://claude.ai/code)_
